internal/payloads/reservation_payload.go

```diff
@@ -28,6 +28,11 @@ type AWSReservationRequest struct {
 		NewName    *string `json:"new_name"`
 		NewBody    *string `json:"new_body"`
 	}
+	LaunchInstance struct {
```

Comment:
Do we need this substructure? I think `AWSReservation` = Reserve an instance ( I agree that the choice of the name is not great, but `launch` and reserve sound kinda same, so I'm not sure if we need nested structure for it.

Reply:
I am not sure it is for launch/reserve an instance, do we need the pubkey for launching an instance? 
Why did lzap add the `Pubkey` struct there? 
I see that he used  `AWSReservation` for the response and not the request 
```golang
type AWSReservationResponse struct {
	*models.AWSReservation
}

```

Reply:
oh yeah, that is probably wrong, I guess it should have been for a Request.

the Pubkey struct wraps interchangeble information of how to get `pubkey_id`, I feel this is bit different case :thinking:

Reply:
```golang
type AWSReservationRequest struct {
	AWSReservation AWSReservation
    Pubkey struct {
		    ExistingID *int64  `json:"existing_id"`
		    NewName    *string `json:"new_name"`
		    NewBody    *string `json:"new_body"`
	    }
}
```
This is what you think?

Reply:
doesn't AWSReservation contain the Pubkey already, I think if we embedd AWSReservation, we can get rid of the Pubkey :)

Reply:
I would imagine not having pubkey in AWSReservation because it is an attribute specific to one job. 
and then it makes sense to use my suggestion, but maybe I am missing something? 
why we want pubkey in AWSReservation?
why not create a type per job?